lib/commons/integrity/check/unique_csv_identifiers.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+require 'csv'
+
+module Commons
+  module Integrity
+    class Check
+      # Check that any values in the given columns are unique within the file
+      #
+      # == Configuration Options
+      # * column_names: the columns containing the IDs (default: ["wikidata", "ms_fb"])
+      # * column_case: "fixed" if the column_name must be exactly as specified (default: "any")
+      class UniqueCSVIdentifiers < Base
+        # @return [Array<Error>]
+        # Errors will be in the category `:unique_csv_identifiers`
+        def errors
+          return [] unless pathname_regex.match? pathname.to_s
+          problematic_values.map do |column_name, value|
+            error(unique_csv_identifiers: "Duplicated #{column_name} value: #{value}")
+          end
+        end
+
+        private
+
+        DEFAULT_COLUMN_NAMES = %w[wikidata ms_fb].freeze
+        DEFAULT_COLUMN_CASE = 'any'
+
+        def csv
+          @csv ||= CSV.read(pathname, headers: true)
+        end
+
+        def pathname_regex
+          Regexp.new(my_config.to_h['pathname_regex'] || '.*')
+        end
+
+        def headers
+          csv.headers
+        end
+
+        def column_names
+          (my_config.to_h['column_names'] || DEFAULT_COLUMN_NAMES).map do |column_name|
+            column_name.send(comparison_conversion)
+          end
+        end
+
+        def comparison_type
+          my_config.to_h['column_case'] || DEFAULT_COLUMN_CASE
+        end
+
+        def comparison_conversion
+          return :to_s if comparison_type == 'fixed'
+          :downcase
+        end
+
```

Comment:
There's a lot of duplication between the above methods and `wikidata_identifiers.rb`. Whilst I don't believe that duplication is _necessarily_ a problem (things can sometimes appear more similar than they actually are, and removing the duplication prematurely can lead to poor coupling), here I think they really are doing the same thing, and I suspect that pushing at least some of the common code up into a common "checking a CSV file" class could be helpful not only in removing the duplication, but in making this code much easier to follow by separating out the more 'structural' code of getting the data we care about, from the specific logic of the check itself.

Reply:
WRT this particular method, I'm 99% confident that we never do — nor would want to do — anything with case-insensitive column names anyway, so we could just simplify by removing this from both places.